Add M365 Data Management agent with MCP and GraphicArt

KatT-AI opened the pull request:
## Summary

Adds a new Teams Toolkit sample (`da-sharepoint-data-manager`) — a declarative Microsoft 365 Copilot agent that helps users organize and find SharePoint/OneDrive content.

- **OneDrive & SharePoint** capability for document search and organization guidance
- **MCP integration** with Microsoft Learn (`learn.microsoft.com/api/mcp`) for searching and fetching official documentation
- **GraphicArt** capability for generating visual diagrams of site structures and folder hierarchies
- **`discourage_model_knowledge`** behavior override to prioritize user data and official docs over general model knowledge

## Changes

- Teams Toolkit lifecycle config (`teamsapp.yml`) + environment and VS Code launch/settings
- Teams app package: manifest (v1.24), declarative agent (v1.5), MCP plugin (v2.1), instructions
- Sample documentation and metadata (README + `assets/sample.json`) with images

KatT-AI committed:
First commit complete v1 of SharePoint Pharma Data Manager, for copilot dev camp.

KatT-AI commented:
First merged complete v1 of SharePoint Pharma Data Manager, for copilot dev camp.A specialized Microsoft Copilot agent that combines Microsoft 365 data management with pharmaceutical regulatory compliance expertise. This agent helps users organize SharePoint content, manage permissions, generate analytics, and access regulatory guidance from FDA, EMA, WHO, and ICH sources. It provides real-time access to pharmaceutical documentation standards, GxP compliance requirements, and clinical trial data management best practices while leveraging Microsoft 365's powerful content management capabilities.

waldekmastykarz reviewed (changes requested):
Thank you for the submission @KatT-AI. I suggest we update the description of the agent. Since it only reads files from OneDrive and SharePoint, and does web search, I suppose that we remove the bits about:

- permission management: no ability to read or change permissions
- filtering by permissions: not possible, given all returned information is already security trimmed
- analytics and reporting: this agent doesn't have access to the necessary information to do this
- data governance: this agent doesn't have access to the necessary information to do this
- Pharmaceutical Regulatory Compliance: unless you upload specific content to your site, this won't work 
 
Please update the description of the agent so that it correctly reflects the agent's capabilities. I'll mark this PR as draft until you've changed it.

KatT-AI committed:
Update README.md

KatT-AI committed:
Update declarativeAgent.json

Changing the prompt

KatT-AI committed:
Update instruction.txt

KatT-AI committed:
Update manifest.json

KatT-AI commented:
Hi @waldekmastykarz , 
Thank you for your suggestions, I committed these changes:

• Removed references to permission management, permission-based filtering, analytics & reporting, and data governance to reflect the agent’s actual capabilities.
• Renamed “Pharmaceutical Regulatory Compliance” to “Pharmaceutical Manufacturing Intelligence” and changed the instruction and the agent tone to emphasize guidance based on publicly available regulatory resources.

Let me know if anything else needs tweaking.

waldekmastykarz commented on samples/da-sharepoint-data-manager/README.md:
```diff
@@ -0,0 +1,81 @@
+- Search for information from trusted sources
+- Up-to-date guidance from official documentation
+
+### Pharmaceutical Manufacturing Intelligence
```
This seems out of scope if the agent is named "Data Manager". Let's remove it because it doesn't seem related to the agent's functionality, not even looking at the agent's conversation starters.

garrytrinder commented:
@KatT-AI are you able to provide an update? When you have resolved the comments, please mark this PR as ready for review, thank you!

Commit:
Address reviewer feedback and update sample to latest standards

- Remove Pharmaceutical Manufacturing Intelligence section from README per reviewer request
- Remove pharma/regulatory references (FDA, EMA, WHO, ICH) from instruction.txt
- Restructure README to match latest pnp/copilot-pro-dev-samples template
- Update tracking image to markdown format per declarative-agent-sample-review checklist
- Update app manifest schema from v1.19 to v1.24
- Update declarative agent schema from v1.3 to v1.5
- Add assets/sample.json metadata file for gallery listing
- Fix Help section issue label from da-qna-graphapi-plugin to da-sharepoint-data-manager
- Clear tenant-specific IDs from env/.env.dev
- Normalize URLs to remove mixed locale prefixes (en-us, en-gb)
- Update contributor GitHub account to KatT-AI for consistency with PR author

Commit:
Add MCP integration, GraphicArt capability, and behavior overrides

- Add ai-plugin.json with Microsoft Learn MCP server (search and fetch)
- Update declarativeAgent.json: add MCP action, GraphicArt capability, behavior_overrides with discourage_model_knowledge, disclaimer, updated conversation starters
- Update instruction.txt: add MCP tool usage guidelines, fallback behavior, Sources section requirement, GraphicArt instructions
- Update manifest.json: schema v1.24
- Update README.md: reflect MCP, GraphicArt, and grounded responses features
- Update sample.json: set API-PLUGIN to Yes, update descriptions for MCP
- Clear .env.dev tenant-specific IDs

garrytrinder commented:
@KatT-AI please review the latest comments, the PR has been marked as draft, mark it as ready to review when you have reviewed the comments.

Commit:
Fix schema_version mismatch in ai-plugin.json and clear tenant IDs

Align schema_version with $schema URL (v2.4 → v2.1) and restore empty
tenant-specific values in .env.dev.

garrytrinder commented:
Merged manually into main — thank you for your contribution, @KatT-AI! 🎉
